### PR TITLE
feat(ingest/dynamodb): add domain arg

### PR DESCRIPTION
**Pull request by dim-ops**
## Checklist

Add domain args to dynamodb

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py [hunk 1]
```diff
@@ -480,3 +492,20 @@ def get_field_type(
...
+                return make_domain_urn(domain)
+
+        return None
+
+    def _get_domain_wu(
+        self, dataset_name: str, entity_urn: str
+    ) -> Iterable[MetadataWorkUnit]:
+        domain_urn = self._gen_domain_urn(dataset_name)
+        if domain_urn:
+            yield from add_domain_to_entity_wu(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:508]: we have a "DomainRegistry" helper, which lets us convert domain names into urns using a lookup against DataHub 

Let's try to reuse that if possible. A good example to reference is the kafka integration's domain support: https://github.com/datahub-project/datahub/blob/1741c07d769f56a9cf066172725384b4e8780839/metadata-ingestion/src/datahub/ingestion/source/kafka.py#L445

Comment (dim-ops) [reply metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:519]: Is it better @hsheth2 ?

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py:513]: yup this looks great!

**Review by hsheth2 — APPROVED** [review]
LGTM

**Comment by anshbansal** [comment]
@dim-ops Please refer to this doc https://datahubproject.io/docs/next/metadata-ingestion/developing/#code-style to fix the lint errors and then we can merge this in

**Comment by dim-ops** [comment]
> @dim-ops Please refer to this doc https://datahubproject.io/docs/next/metadata-ingestion/developing/#code-style to fix the lint errors and then we can merge this in

I format my code, is it better ? @anshbansal

**Comment by anshbansal** [comment]
> > @dim-ops Please refer to this doc https://datahubproject.io/docs/next/metadata-ingestion/developing/#code-style to fix the lint errors and then we can merge this in
> 
> I format my code, is it better ? @anshbansal

@dim-ops The CI was failing. We have made some improvements to the CI to make it more robust. I have merged in `master` branch and when CI is green it should be good to merge

**Pull request merged**